src/Illuminate/Foundation/Testing/TestResponse.php

```diff
@@ -631,30 +631,39 @@ public function assertJsonCount(int $count, $key = null)
     }
 
     /**
-     * Assert that the response has the given JSON validation errors for the given keys.
+     * Assert that the response has the given JSON validation errors.
      *
-     * @param  string|array  $keys
+     * @param  string|array  $errors
      * @return $this
      */
-    public function assertJsonValidationErrors($keys)
+    public function assertJsonValidationErrors($errors)
     {
-        $keys = Arr::wrap($keys);
+        $errors = Arr::wrap($errors);
 
-        PHPUnit::assertNotEmpty($keys, 'No keys were provided.');
+        PHPUnit::assertNotEmpty($errors, 'No validation errors were provided.');
 
-        $errors = $this->json()['errors'] ?? [];
+        $jsonErrors = $this->json()['errors'] ?? [];
 
-        $errorMessage = $errors
+        $errorMessage = $jsonErrors
                 ? 'Response has the following JSON validation errors:'.
-                        PHP_EOL.PHP_EOL.json_encode($errors, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE).PHP_EOL
+                        PHP_EOL.PHP_EOL.json_encode($jsonErrors, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE).PHP_EOL
                 : 'Response does not have JSON validation errors.';
 
-        foreach ($keys as $key) {
-            PHPUnit::assertArrayHasKey(
-                $key,
-                $errors,
-                "Failed to find a validation error in the response for key: '{$key}'".PHP_EOL.PHP_EOL.$errorMessage
-            );
+        foreach ($errors as $key => $value) {
+            if (is_int($key)) {
+                PHPUnit::assertArrayHasKey(
+                    $value,
+                    $jsonErrors,
+                    "Failed to find a validation error in the response for key: '{$value}'".PHP_EOL.PHP_EOL.$errorMessage
+                );
+            } else {
+                PHPUnit::assertArraySubset(
```

Comment:
Please do not introduce a new usage of `assertArraySubset` in Laravel. This is deprecated in PHPUnit 8 and we should avoid making adding more uses of it.

Reply:
@driesvints Thanks. I just read the the deprecated post. It looks like there is no alternative.

hmm, unfortunately, now there isn't a clean way to check if a key and value matches within an array in PHPUnit and changes to use PHP here would be ugly by checking if key is set in `jsonErrors` then if the error message matches.  Might have to table this until PHPUnit's issue/what you decide for `assertJson` is sorted out. Thoughts?

Reply:
If we can't find an alternative we're going to backport it into the framework directly but until we figure it out it's best to avoid adding more uses.

Reply:
@driesvints if you are looking for a quick way out, I ported them to a phpunit extension https://packagist.org/packages/dms/phpunit-arraysubset-asserts

Reply:
@rdohms ah thanks. I'll put this on my todo list. Thanks for sharing 👍